src/parser/DASHTree.cpp

```diff
@@ -600,11 +600,13 @@ start(void *data, const char *el, const char **attr)
           {
             if (strcmp(value, "subtitle") == 0)
               dash->current_adaptationset_->type_ = DASHTree::SUBTITLE;
-            //Legacy compatibility
-            if (strcmp(value, "forced") == 0)
-              dash->current_adaptationset_->forced_ = true;
-            if (strcmp(value, "main") == 0)
-              dash->current_adaptationset_->default_ = true;
+            else if (dash->current_adaptationset_->type_ == DASHTree::SUBTITLE) {
```

Comment:
We use to have brackets at new line (see clang-format file)
Beside this your change does only work, if `if (strcmp(value, "subtitle") == 0)` is at first place.

You'll have to store forced and main in local variables and set them later.

Reply:
hopefully fixed it